apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -500,34 +505,41 @@ def _set_reader_coord(
         # Note: slice(None, None) matches the is_slice_of part, unless we also check the dim-type
         # part.
         if (is_slice_of(coord, str) or is_slice_of(coord, bytes)) and (
-            dim.dtype == "str" or dim.dtype == "bytes"
+            pa.types.is_large_string(dim.type)
+            or pa.types.is_large_binary(dim.type)
+            or pa.types.is_string(dim.type)
+            or pa.types.is_binary(dim.type)
         ):
             _util.validate_slice(coord)
             # Figure out which one.
-            dim_type: Union[Type[str], Type[bytes]] = type(dim.domain[0])
+            dim_type: Union[Type[str], Type[bytes]] = type(self.domain[dim_idx][0])
```

Comment:
Since we look at `self.domain[dim_idx]` a bunch in this code, would it make sense to pull it out near the top of the function and store it in a variable?